File: Golang/LongestPalindromicSubstring/LongestPalindromicSubstring_test.go
package LongestPalindromicSubstring

import "testing"

func TestLongestPalindrome(t *testing.T) {

	s := longestPalindrome("babad")
	if s == "bab" || s == "aba" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if longestPalindrome("cbbd") == "bb" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if longestPalindrome("babaddtattarrattatddetartrateedredividerb") == "ddtattarrattatdd" {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/LongestPalindromicSubstringTest.py
from unittest import TestCase

from LongestPalindromicSubstring import LongestPalindromicSubstring


class TestLongestPalindromicSubstring(TestCase):
    def test_longestPalindrome(self):
        lps = LongestPalindromicSubstring()

        s = lps.longestPalindrome("babad")
        self.assertTrue(s == "bab" or s == "aba")

        self.assertEqual(lps.longestPalindrome("cbbd"), "bb")

        self.assertEqual(lps.longestPalindrome("babaddtattarrattatddetartrateedredividerb"), "ddtattarrattatdd")
